components/builder-db/src/models/origin.rs

```diff
@@ -119,6 +122,42 @@ impl Origin {
             .execute(conn)
     }
 
+    pub fn delete(origin: &str, conn: &PgConnection) -> QueryResult<()> {
+        Counter::DBCall.increment();
+        conn.transaction::<_, Error, _>(|| { 
+            // This is ugly but should be relatively safer than some alternatives
+            // It turns out we don't have cascade on delete in the schema for the 
+            // origin across the 12 tables it uses. We can add it but it is a bit
+            // scary. It would be a nuclear hammer.
+            diesel::delete(
+                origin_channels::table
+                    .filter(origin_channels::origin.eq(origin)))
+                    .execute(conn)?;
+            diesel::delete(
+                origin_secret_keys::table
+                    .filter(origin_secret_keys::origin.eq(origin)))
+                    .execute(conn)?;
+            diesel::delete(
+                origin_public_keys::table
+                    .filter(origin_public_keys::origin.eq(origin)))
+                    .execute(conn)?;
+            diesel::delete(
+                origin_members::table
```

Comment:
I think we may want to also make sure that there are also no members other than the owner in the origin before we trigger the delete

Reply:
I can get this in tomorrow morning. Is this something we definitely want? This deletion will remove any membership associated with the origin being terminated.

I don't have a preference either way mostly curious your thoughts on it.

Reply:
I'm mostly concerned about any undefined behavior if there are other members in the origin. If things look clean after the deletion then go for it. We can also add some warning message(s) in the cli/UI later.

Reply:
Yep, I validated this locally, there may be other undiscovered behavior here but WRT members, everything appears clean in the UI for the owner as well as the subordinate members.